sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -15,59 +15,64 @@ export 'package:sky/src/painting/text_painter.dart';
 // differently. To work around this problem for now, we round fractional pixel
 // values up to the nearest whole pixel value. The right long-term fix is to do
 // layout using fixed precision arithmetic.
-double _applyFloatingPointHack(double layoutValue) {
+double applyFloatingPointHack(double layoutValue) {
   return layoutValue.ceilToDouble();
 }
 
 /// A render object that displays a paragraph of text
 class RenderParagraph extends RenderBox {
 
-  RenderParagraph(TextSpan text) : _textPainter = new TextPainter(text) {
+  RenderParagraph(TextSpan text) : textPainter = new TextPainter(text) {
     assert(text != null);
   }
 
-  TextPainter _textPainter;
+  // Exposed for use by subclasses.
+  TextPainter textPainter;
 
   BoxConstraints _constraintsForCurrentLayout; // when null, we don't have a current layout
 
   /// The text to display
-  TextSpan get text => _textPainter.text;
+  TextSpan get text => textPainter.text;
   void set text(TextSpan value) {
-    if (_textPainter.text == value)
+    if (textPainter.text == value)
       return;
-    _textPainter.text = value;
+    textPainter.text = value;
     _constraintsForCurrentLayout = null;
     markNeedsLayout();
   }
 
-  void _layout(BoxConstraints constraints) {
+  // Can be overridden by subclasses.
+  bool allowLineWrap() => true;
```

Comment:
Why not take this value as a constructor parameter and provide a getter and a setter?  As written, the value is neither final nor is there a notification pathway for when it changes.  We should either make it final (e.g., by taking the value as a constructor parameter and not providing a setter) or have a notification pathway for it changing (e.g., by having a setting that calls markNeedsLayout as appropriate).